File: NaviTest/NaviTest/PinkieNavigation/NavigationController/PinkieNavigationDelegateImp.m
//
//  PinkieNavigationDelegateImp.m
//  NaviTest
//
//  Created by jiazhiyi on 2019/4/9.
//  Copyright © 2019 jiazhiyi. All rights reserved.
//

#import "PinkieNavigationDelegateImp.h"
#import "PinkiePushAnimatedTransitioning.h"
#import "PinkiePopAnimatedTransitioning.h"

@implementation PinkieNavigationDelegateImp

- (id<UIViewControllerAnimatedTransitioning>)navigationController:(UINavigationController *)navigationController animationControllerForOperation:(UINavigationControllerOperation)operation fromViewController:(UIViewController *)fromVC toViewController:(UIViewController *)toVC
{
    if (operation == UINavigationControllerOperationPush) {
        return [[PinkiePushAnimatedTransitioning alloc] init];
        
    } else if (operation == UINavigationControllerOperationPop) {
        return [[PinkiePopAnimatedTransitioning alloc] init];
    } else {
        return nil;
    }
}

- (id<UIViewControllerInteractiveTransitioning>)navigationController:(UINavigationController *)navigationController interactionControllerForAnimationController:(id<UIViewControllerAnimatedTransitioning>)animationController
{
    return self.transitioning.isInteracting ? self.transitioning : nil;
}

@end

File: CustomNavigationBar/CustomNavigationBar/PinkieNavigation/NavigationController/PinkieNavigationDelegateImp.swift
//
//  PinkieNavigationDelegateImp.swift
//  CustomNavigationBar
//
//  Created by jiazhiyi on 2019/4/4.
//  Copyright © 2019 jiazhiyi. All rights reserved.
//

import UIKit

class PinkieNavigationDelegateImp: NSObject & UINavigationControllerDelegate {
    
    var popTransition: PinkieNavPopTransitioning
    
    init(popTransition: PinkieNavPopTransitioning) {
        self.popTransition = popTransition
    }
    
    func navigationController(_ navigationController: UINavigationController, willShow viewController: UIViewController, animated: Bool) {
        
        print("willShowViewController = \(animated)")
        

    }
    
    func navigationController(_ navigationController: UINavigationController, didShow viewController: UIViewController, animated: Bool) {
        
    }
    
    func navigationController(_ navigationController: UINavigationController, animationControllerFor operation: UINavigationController.Operation, from fromVC: UIViewController, to toVC: UIViewController) -> UIViewControllerAnimatedTransitioning? {
        switch operation {
        case .push:
            return PinkiePushAnimatedTransitioning()
        case .pop:
            return PinkiePopAnimatedTransitioning()
        default:
            return nil
        }
    }

    func navigationController(_ navigationController: UINavigationController, interactionControllerFor animationController: UIViewControllerAnimatedTransitioning) -> UIViewControllerInteractiveTransitioning? {
        return self.popTransition.isInteracting ? self.popTransition : nil
    }
}
